File: extract_pdf.js
const pdfParseModule = require('pdf-parse');
const pdfParse = pdfParseModule.default || pdfParseModule;
const fs = require('fs');
const path = require('path');

async function extractPdf(filePath, outName) {
  const buf = fs.readFileSync(filePath);
  const data = await pdfParse(buf);
  fs.writeFileSync(path.join(__dirname, outName), data.text, 'utf8');
  console.log(`✅ Extracted ${outName} — ${data.numpages} pages, ${data.text.length} chars`);
  // Print first 3000 chars as preview
  console.log('\n--- PREVIEW ---\n');
  console.log(data.text.substring(0, 3000));
  console.log('\n--- END PREVIEW ---\n');
  return data.text;
}

(async () => {
  console.log('=== LISTENING PDF ===');
  await extractPdf('assets/listening/test1/questions.pdf', 'listening_raw.txt');
  console.log('\n=== READING PDF ===');
  await extractPdf('assets/reading/test1.pdf', 'reading_raw.txt');
})();

File: extract_pdf.py
import pdfplumber, json, re, sys
sys.stdout.reconfigure(encoding='utf-8') if hasattr(sys.stdout, 'reconfigure') else None

def extract(pdf_path, out_path):
    lines = []
    with pdfplumber.open(pdf_path) as pdf:
        for page in pdf.pages:
            text = page.extract_text()
            if text:
                lines.append(text)
    full = "\n".join(lines)
    with open(out_path, "w", encoding="utf-8") as f:
        f.write(full)
    print(f"OK {out_path}: {len(full)} chars")
    print("--- PREVIEW ---")
    print(full[:4000])
    print("--- END ---")
    return full

extract("assets/listening/test1/questions.pdf", "listening_raw.txt")
print("\n\n========\n\n")
extract("assets/reading/test1.pdf", "reading_raw.txt")
